### PR TITLE
zephyr: some logging related updates

### PR DESCRIPTION
**Pull request by mike-scott**
- INPLACE_PROCESS -> IMMEDIATE change following upstream refactor
- Establish log level for MCUBOOT specifically
- Settings to ensure other subsys don't log and cause bloat

### CODE REVIEW

**Comment by mike-scott** [comment]
Adding @lemrey and @utzig due to their familiarity with the last set of Logging changes.

Also, I have 1 more change coming which will save about 3K flash but requires a Zephyr-side PR to be merged first.  The net of that will be that the ROM size for Zephyr-based MCUBOOT builds should almost be the same as we were prior to Logger migration.

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -225,6 +225,11 @@ config MCUBOOT_DEVICE_CPU_CORTEX_M0
 	default n
 	select SW_VECTOR_RELAY if !CPU_CORTEX_M0_HAS_VECTOR_TABLE_REMAP
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: Put just after `endif # MCUBOOT_SERIAL`, within menu `MCUBoot settings`. Thanks to that the option will be in this menu.

[the remark this answers is not in the document]
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: Inside `if BOOT_HAVE_LOGING` clause as well.

**Comment by utzig** [comment]
@nvlsianpu 133b002749d88814d5f510907e56eca9a3e3a516 is missing your sob.

**Comment by nvlsianpu** [comment]
^^fixed

**Comment by mike-scott** [comment]
@nvlsianpu Thank you for updating this!  Work got very busy and since 1.3 was tagged, this fell down in priority.

**Comment by nvlsianpu** [comment]
@nordic-krch revisit

**Pull request merged**